examples/charts/rendering-spec/src/index.js

```diff
@@ -0,0 +1,124 @@
+import ChartsEmbedSDK from "@mongodb-js/charts-embed-dom";
+import "@alenaksu/json-viewer";
+
+const sdk = new ChartsEmbedSDK({
+  baseUrl: "https://charts.mongodb.com/charts-embedding-examples-wgffp", // Optional: ~REPLACE~ with the Base URL from your Embed Chart dialog
+});
+
+const chart = sdk.createChart({
+  chartId: "735cfa75-15b8-483a-bc2e-7c6659511c7c", // Optional: ~REPLACE~ with the Chart ID from your Embed Chart dialog
+  height: "700px",
+});
+
+async function renderChart() {
+  await chart.render(document.getElementById("chart"));
+
+  /*
+    Collecting all the data from the form and based on it preparing the rendering spec
+    that will be sent to chart.setRenderingSpecOverride() to apply customizations on the embedded chart
+  */
+  document.forms["form"].addEventListener("submit", async function (e) {
+    e.preventDefault();
+
+    // get all the values from the form
+    const formMarkup = new FormData(e.target);
+    const formData = Object.fromEntries(formMarkup);
+
+    const { title, description, labelX, labelY, labelSize, colorPalette } =
+      formData;
+
+    // prepare the rendering spec document
+    const renderingSpec = {
+      version: 1,
+      ...(title ? { title } : {}),
+      ...(description ? { description } : {}),
+    };
+
+    if (labelX || labelY) {
+      renderingSpec.channels = {
+        ...(labelX ? { x: { labelOverride: labelX } } : {}),
+        ...(labelY ? { y: { labelOverride: labelY } } : {}),
+      };
+    }
+
+    if (labelSize || colorPalette) {
+      renderingSpec.options = {
+        ...(labelSize ? { labelSize: parseInt(labelSize) } : {}),
+        ...(colorPalette
+          ? {
+              colorPalette: {
+                type: "discrete",
+                value: [colorPalette],
+              },
+            }
+          : {}),
+      };
+    }
+
```

Comment:
I was wondering if it would be more beneficial to add an example for a table chart with conditional formatting? Perhaps we can create a tech debt ticket and we can pick it up later.

Reply:
I agree that it will be great to add more examples, tho we can't give an example for everything that the renderingSpec can do. We have a code snippet in the docs for table chart and conditional formatting with 3-4 rules.